Service/ContentTypeService.php

```diff
@@ -426,10 +430,30 @@ public function getAllTypes()
         return implode(',', array_keys($this->contentTypeArrayByName));
     }
 
-    public function contentTypeFromJson(string $json, Environment $environment): ContentType
+
+    public function updateFromJson(ContentType $contentType, string $json, bool $isDeleteExitingTemplates, bool $isDeleteExitingViews): void
+    {
+        $this->deleteFields($contentType);
+        if ($isDeleteExitingTemplates) {
+            $this->deleteTemplates($contentType);
+        }
+        if ($isDeleteExitingViews) {
+            $this->deleteViews($contentType);
+        }
+
+        $environment = $contentType->getEnvironment();
+        if (! $environment instanceof Environment) {
+            throw new NotFoundHttpException('Environment not found');
+        }
+
+        $updatedContentType = $this->contentTypeFromJson($json, $environment, $contentType);
+        $this->importContentType($updatedContentType);
```

Comment:
The current strategy is to remove everything unused from the database. Then update from the json (or only the properties from the json). By refactoring (my other comments) this behaviour would be more clear. 

I have a question however. Why not create a ContentTypeFromJson with the new $json file. Then compare the currenct ContentType with the newly created one $oldContentType-->merge(Contenttype $ct) and then persist this updated representation of the contenttype in the database?

That would solve the inconsistent db state created at line 436.